src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -345,73 +343,54 @@ export function makeUtxoEngineState(
       branch = 0,
       forceIndex
     }): Promise<EdgeFreshAddress> {
-      const walletPurpose = currencyFormatToPurposeType(
-        walletInfo.keys.privateKeyFormat
-      )
-      if (walletPurpose === BIP43PurposeTypeEnum.Segwit) {
-        const {
-          address: publicAddress,
-          nativeBalance = '0',
-          legacyAddress
-        } = await internalGetFreshAddress({
-          ...commonArgs,
-          format: getCurrencyFormatFromPurposeType(
-            BIP43PurposeTypeEnum.WrappedSegwit
-          ),
-          forceIndex,
-          changeIndex: branch
-        })
+      const { privateKeyFormat } = walletInfo.keys
+
+      // Airbitz wallets only use branch 0
+      if (privateKeyFormat === 'bip32') branch = 0
+
+      const {
+        address: publicAddress,
+        nativeBalance = '0',
+        legacyAddress
+      } = await internalGetFreshAddress({
+        ...commonArgs,
+        format: privateKeyFormat,
+        forceIndex,
+        changeIndex: branch
+      })
 
-        const {
-          address: segwitAddress,
-          nativeBalance: segwitNativeBalance = '0'
-        } = await internalGetFreshAddress({
-          ...commonArgs,
-          format: getCurrencyFormatFromPurposeType(BIP43PurposeTypeEnum.Segwit),
-          forceIndex,
-          changeIndex: branch
-        })
+      const freshAddress: EdgeFreshAddress = {
+        publicAddress,
+        nativeBalance,
+        legacyAddress:
+          legacyAddress !== publicAddress ? legacyAddress : undefined,
+        // Legacy address is just a different encoding of the standard public address
+        // and therefore would have the same balance
+        legacyNativeBalance:
+          legacyAddress !== publicAddress ? nativeBalance : undefined
+      }
 
-        return {
-          nativeBalance,
-          publicAddress,
-          legacyNativeBalance:
-            legacyAddress !== publicAddress ? nativeBalance : undefined,
-          // Legacy address is just a different encoding of the standard public address
-          // and therefore would have the same balance
-          legacyAddress:
-            legacyAddress !== publicAddress ? legacyAddress : undefined,
-          segwitAddress,
-          segwitNativeBalance
-        }
-      } else {
-        // Airbitz wallets only use branch 0
-        if (walletPurpose === BIP43PurposeTypeEnum.Airbitz) {
-          branch = 0
-        }
+      // Exclude the privateKeyFormat because it's covered by 'publicAddress'
+      const supportedFormats = getSupportedFormats(
+        pluginInfo.engineInfo,
+        walletInfo.keys.privateKeyFormat
+      ).filter(format => format !== privateKeyFormat)
 
-        const {
-          address: publicAddress,
-          nativeBalance = '0',
-          legacyAddress
-        } = await internalGetFreshAddress({
+      for (const format of supportedFormats) {
```

Comment:
I don't understand why this is a loop, if we only care about bip84. Would `if (supportedFormats.includes('bip84'))` do the same thing, since any other value seems to be a no-op?

Reply:
You're quite right. I think I over-thought the code infrastructure here; I wrote as if any other supported addresses may be handled by the loop in the future, but there is no handlers for these cases anyway, so perhaps simple if statements instead a of a loop would be more straight-forward.

Reply:
I've made a change to no longer uselessly derive formats which don't impact the return value. I've left the loop in place as infrastructure to add support for other formats for the bip49 private key format (e.g. bech32m/taproot/bip350).